src/OrchardCore/OrchardCore.Data.YesSql.Abstractions/IndexesServiceCollectionExtensions.cs

```diff
@@ -0,0 +1,19 @@
+using Microsoft.Extensions.DependencyInjection;
+using YesSql.Indexes;
+
+namespace OrchardCore.Data;
+
+public static class IndexesServiceCollectionExtensions
+{
+    public static IServiceCollection AddIndexProvider<TIndexProvider>(this IServiceCollection services)
+        where TIndexProvider : class, IIndexProvider
+    {
+        return services.AddSingleton<IIndexProvider, TIndexProvider>();
+    }
+
+    public static IServiceCollection AddScopedIndexProvider<TIndexProvider>(this IServiceCollection services)
```

Comment:
I like the `AddIndexProvider` but not this one because naming is confusing little bit it registers `IIndexProvider` as scoped service or registers `IScopedIndexProvider`

Reply:
sure. I renamed it

Reply:
I meant the API name itself not the parameter ;) let us hear the folks feedback

Reply:
I meant the API name itself not the parameter ;) let us hear the folks feedback